pallets/utxo/src/lib.rs

```diff
@@ -230,12 +230,53 @@ pub mod pallet {
         }
     }
 
+    #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+    #[derive(Clone, Encode, Decode, Eq, PartialEq, PartialOrd, Ord, RuntimeDebug, Hash)]
+    pub enum TxHeaderAndExtraData {
+        NormalTx {
+            // Normal token ID
+            id: TokenId,
+        },
+        ConfidentialTx, /* not implemented yet */
+        Nft {
+            id: TokenId,
+            data_hash: [u8; 32],
+            data_url: String,
+            creator_pkh: H256,
+        },
+    }
+
+    impl TxHeaderAndExtraData {
+        pub fn id(&self) -> Option<TokenId> {
+            match self {
+                Self::NormalTx { id } => Some(*id),
+                Self::Nft { id, .. } => Some(*id),
+                _ => None,
+            }
+        }
+
+        pub fn is_normal(&self) -> bool {
+            match self {
+                Self::NormalTx { .. } => true,
+                _ => false,
+            }
+        }
+
+        pub fn is_nft(&self) -> bool {
+            match self {
+                Self::Nft { .. } => true,
+                _ => true,
+            }
+        }
+    }
+
     /// Output of a transaction
     #[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
     #[derive(Clone, Encode, Decode, Eq, PartialEq, PartialOrd, Ord, RuntimeDebug, Hash)]
+    #[repr(C)]
```

Comment:
As explained previously, we should never use `repr(C)` anywhere, because this is architecture dependent. Alternatively we should use some serialization codec like SCALE or json or whatever. If you create a transaction on x86_64, then you move it to Raspberry Pi, everything will fail if you use `repr(C)`.

Reply:
You are right, I just forgot to remove that after our call. Done.